File: internal/analyze/java/testdata/inheritance.java
package com.acme.app;

class Ledger extends Base implements Auditable, Printer {
    // inheritance 3 — Base, Auditable, Printer, one occurrence each
}

interface Auditable extends Named, Timestamped {
    // inheritance 2
}

enum Level implements Auditable {
    LOW, HIGH                              // inheritance 1, local_variable 0 (enum constants)
}

record Money(int amount, String currency) implements Comparable<Money> {
    // inheritance 1 (type arguments do not add), local_variable 2 (the components)
    public int compareTo(Money other) {
        return amount - other.amount;
    }
}

class Factory {
    Runnable r = new Runnable() {          // inheritance 1 (anonymous class, on `Runnable`),
        public void run() {                // local_variable 1 (the field r), lambda 0
        }
    };
}

sealed interface Shape permits Circle {
    // inheritance 0 — `permits` is not an edge; Circle lives elsewhere in the package
}

File: internal/analyze/kotlin/testdata/inheritance.kt
class Ledger(private val repo: Repo, clock: Clock) : Base(), Auditable, Printer by ConsolePrinter()
// inheritance 3, local_variable 1 (repo; clock is a plain parameter)

interface Auditable : Named, Timestamped                 // inheritance 2
object Registry : Auditable                              // inheritance 1
class Bare                                               // inheritance 0
class Outer {
    inner class In : Base()                              // inheritance 1, on Outer
}
enum class Color : Named {                               // inheritance 1
    RED
}
class X : Generic<String>()                              // inheritance 1
fun anonymous() {
    val r = object : Runnable {                          // inheritance 1, local_variable 1
        override fun run() {}
    }
}
